test/test-mongocrypt-crypto.c

```diff
@@ -19,325 +19,304 @@
+            !test->ignore_encrypt_failure,
+            "Duplicate field 'ignore_encrypt_failure' in test");
+         ASSERT (BSON_ITER_HOLDS_BOOL (iter));
+         ASSERT_OR_PRINT_MSG (bson_iter_bool (iter),
+                              "value of 'ignore_encrypt_failure' must be true");
+         test->ignore_encrypt_failure = true;
+      }
 
-   mongocrypt_t *crypt;
-   crypt = mongocrypt_new ();
+#define STR_FIELD(Name)                                            \
+   else if (!strcmp (field, #Name))                                \
+   {                                                               \
+      ASSERT_OR_PRINT_MSG (!test->Name,                            \
+                           "Duplicate field '" #Name "' in test"); \
+      ASSERT (BSON_ITER_HOLDS_UTF8 (iter));                        \
+      test->Name = bson_strdup (bson_iter_utf8 (iter, NULL));      \
+   }
 
-   for (test = tests; test->testname != NULL; test++) {
-      bool ret;
-      _mongocrypt_buffer_t key;
-      _mongocrypt_buffer_t iv;
-      _mongocrypt_buffer_t plaintext;
-      _mongocrypt_buffer_t ciphertext;
-      _mongocrypt_buffer_t plaintext_got;
-      _mongocrypt_buffer_t ciphertext_got;
-      mongocrypt_status_t *status;
-      uint32_t bytes_written;
+      STR_FIELD (name)
+      STR_FIELD (encrypt_error)
+      STR_FIELD (decrypt_error)
+
+#undef STR_FIELD
+
+#define HEXBUF_FIELD(Name)                                         \
+   else if (!strcmp (field, #Name))                                \
+   {                                                               \
+      ASSERT_OR_PRINT_MSG (!test->Name.data,                       \
+                           "Duplicate field '" #Name "' in test"); \
+      ASSERT (BSON_ITER_HOLDS_UTF8 (iter));                        \
+      const char *value = bson_iter_utf8 (iter, NULL);             \
+      _mongocrypt_buffer_copy_from_hex (&test->Name, value);       \
+      ASSERT (strlen (value) == (test->Name.len * 2));             \
+   }
 
-      if (!_aes_ctr_is_supported_by_os) {
-         printf ("Common Crypto with no CTR support detected. Skipping.");
-         return;
-      }
+      HEXBUF_FIELD (key)
+      HEXBUF_FIELD (aad)
+      HEXBUF_FIELD (iv)
+      HEXBUF_FIELD (plaintext)
+      HEXBUF_FIELD (ciphertext)
+#undef HEXBUF_FIELD
+   }
 
-      printf ("Begin test '%s'.\n", test->testname);
+   ASSERT_OR_PRINT_MSG (test->name, "Missing field 'name'");
+   ASSERT_OR_PRINT_MSG (test->algo, "Missing field 'algo'");
+   ASSERT_OR_PRINT_MSG (test->key.data, "Missing field 'key'");
+   if (test->algo == _mcFLE2Algorithm ()) {
+      ASSERT_OR_PRINT_MSG (
+         test->aad.len == 0,
+         "Unexpected value in field 'aad' for cipher without MAC");
+   } else {
+      ASSERT_OR_PRINT_MSG (test->aad.data, "Missing field 'aad'");
+   }
+   ASSERT_OR_PRINT_MSG (test->iv.data, "Missing field 'iv'");
+   ASSERT_OR_PRINT_MSG (test->plaintext.data || test->encrypt_error,
```

Comment:
```suggestion
   ASSERT_OR_PRINT_MSG (test->plaintext.data || test->decrypt_error,
```

IIUC, `plaintext` is required if `encrypt_error` is present.